pkg/chain/ethlike/balance_monitor.go

```diff
@@ -22,17 +25,26 @@ func NewBalanceMonitor(balanceSource BalanceSource) *BalanceMonitor {
 // Observe starts a process which checks the address balance with the given
 // tick and triggers an alert in case the balance falls below the
 // alert threshold value.
+// The balance check will be retried in case of an error up to the retry timeout.
 func (bm *BalanceMonitor) Observe(
 	ctx context.Context,
 	address Address,
 	alertThreshold *Token,
 	tick time.Duration,
+	retryTimeout time.Duration,
```

Comment:
Non-blocking but shouldn't we validate `retryTimeout < tick` so that we do not have two checks overlapping?

Reply:
Yeah, good idea.

Reply:
On the other hand, don't we want to leave it to the implementation using this funcion?